Sources/Misc/AnyDecodable.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  AnyDecodable.swift
+//
+//  Created by Nacho Soto on 5/11/22.
+
+import Foundation
+
+/// Type erased `Any` that conforms to `Decodable`
+enum AnyDecodable {
+
+    case string(String)
+    case int(Int)
+    case double(Double)
+    case bool(Bool)
+    case object([String: AnyDecodable])
+    case array([AnyDecodable])
+    case null
+
+}
+
+extension AnyDecodable: Decodable {
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+
+        if container.decodeNil() {
+            self = .null
+        } else if let value = try? container.decode(String.self) {
+            self = .string(value)
+        } else if let value = try? container.decode(Int.self) {
```

Comment:
I shouldn't be surprised, yet I am- if you pass in `2.5` it won't try to decode as an `Int`, it will work as a `Double`. That's neat.

Reply:
Types are nice